src/config.ml

```diff
@@ -2125,12 +2155,13 @@ let client_generate_connect_options t =
   let excerpt =
     Conf_map.filter
       (function
-        | B (Cipher, _) -> true
+        | B (Cipher, _) -> false (* TODO: only relevant if no NCP is used *)
```

Comment:
this TODO should be addressed... AFAICT we have two modes of operation: tls-auth or static key mode. in the former, we do NCP and do not need the cipher in the push_request, in the latter we need it.

question: do we need to support "connecting a client with NCP to a server without NCP?" -- but I struggle to understand how we can distinguish the server not having NCP (and maybe it's not worth it, since NCP is around for > 4 years now)

Reply:
done in 1bc5456